packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/InteractionLayer.js

```diff
@@ -28,36 +28,35 @@ function InteractionLayer ({ activeDrawingTask, svg }) {
     svgEvent.x = x
     svgEvent.y = y
     const svgEventOffset = svgEvent.matrixTransform(svg.getScreenCTM().inverse())
-
     return svgEventOffset
   }
 
-  function onPointerDown (event) {
-    const activeTool = activeDrawingTask.activeTool
+  function createMark (event) {
+    const { activeTool } = activeDrawingTask
     const activeMark = activeTool.createMark({
       id: cuid(),
       toolIndex: activeDrawingTask.activeToolIndex
     })
-    activeMark.setCoordinates(convertEvent(event))
+    activeMark.initialPosition(convertEvent(event))
     setActiveMark(activeMark)
+    setCreating(true)
   }
 
   function onPointerMove (event) {
-    if (activeMark) {
-      activeMark.setCoordinates(convertEvent(event))
-    }
+    creating && activeMark.initialDrag(convertEvent(event))
   }
 
-  function onPointerUp (event) {
-    if (activeMark) {
-      activeMark.setCoordinates(convertEvent(event))
+  function onPointerUp () {
+    setCreating(false)
+    if (!activeMark.isValid) {
+      const { activeTool } = activeDrawingTask
+      activeTool.deleteMark(activeMark)
     }
-    setActiveMark(null)
   }
 
   return (
     <g
-      onPointerDown={onPointerDown}
+      onPointerDown={createMark}
```

Comment:
Having descriptive names of what the handler is doing is fine with me, but I'd like some consistency. If we're using more descriptive names for the handlers, then I'd like `onPointerMove` and `onPointerUp` renamed as well, perhaps to `dragMark` and `validateMark` perhaps?

Reply:
I think the original names for the mark creation methods in PFE were `initStart`, `initDrag` and `initEnd`. We could use those?